File: HW 6 (Multithreading)/multithreading/src/main/java/com/example/multithreading/database/AppDatabase.java
package com.example.multithreading.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Contact.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    public abstract ContactDao getContactDao();
    private static volatile AppDatabase appDatabase;

    public static AppDatabase getAppDatabaseInstance(Context context){
        if(appDatabase == null){
            appDatabase = Room.databaseBuilder(context.getApplicationContext(),
                    AppDatabase.class, "CONTACT_TABLE")
                    .build();
        }
        return appDatabase;
    }
}

File: HW 9 (ContentProvider, ContentResolver)/contentprovider/src/main/java/com/example/contentprovider/database/AppDatabase.kt
package com.example.contentprovider.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

@Database(entities = arrayOf(Contact::class), version = 1, exportSchema = false)
@TypeConverters(EnumTypeConverter::class)
abstract class AppDatabase : RoomDatabase() {

    abstract fun getContactDao(): ContactDao

    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null

        @JvmStatic
        fun getAppDatabaseInstance(context: Context): AppDatabase {
            if (INSTANCE == null) {
                INSTANCE = Room.databaseBuilder(
                        context.applicationContext,
                        AppDatabase::class.java,
                        "CONTACT_TABLE"
                )
                        .allowMainThreadQueries()
                        .build()
            }
            return INSTANCE as AppDatabase
        }
    }
}